prog/kube-peers/main.go

```diff
@@ -37,9 +38,9 @@ func getKubePeers() ([]string, error) {
 	for _, peer := range nodeList.Items {
 		var internalIP, externalIP string
 		for _, addr := range peer.Status.Addresses {
-			if addr.Type == "InternalIP" {
+			if addr.Type == "InternalIP" && addressLooksOK(addr.Address) {
```

Comment:
I scanned the Kubernetes code to try to find where these addresses come from.  There are various implementations, from AWS, Rackspace, Openstack, etc.  There is no consistency, but they do all seem likely to generate a numeric IP address rather than a name.